cpp/include/raft/linalg/cusolver_wrappers.h

```diff
@@ -358,7 +359,8 @@ inline cusolverStatus_t cusolverDnsyevdx(  // NOLINT
 template <typename T>
 cusolverStatus_t cusolverDngesvd_bufferSize(  // NOLINT
   cusolverDnHandle_t handle, int m, int n, int *lwork) {
-  if (typeid(T) == typeid(float)) {
+  if (std::is_same<std::remove_cv<std::remove_reference<T>::type>::type,
+                   float>::value) {
```

Comment:
Bit more concise and doing the same thing. 
```suggestion
  if (std::is_same<std::decay<T>::type, float>::value) {
```

Reply:
Thanks, change made.